### DISCUSSION TITLE
Table component - bad column order

### DISCUSSION THREAD

**Discussion by gaborbatori** [Budibase/budibase#14632] [Help]
Please look at the followings!
I have a Data provider which loads data from MySQL table in order to have a Dynamic filter above the table.
The table datasource is the Data Provider. By default, the col order is right, but when i hide or rename, or change anything with the cols, the display column suddenly jumps to the last position, despite it is shown as unchangeable display column (as set in the data sources tab in the application).

<img width="1448" alt="image" src="https://github.com/user-attachments/assets/3144eb3e-d9b0-4e46-b9bf-e601390202ee">

If I set the datasource to the DB table directly, all goes well...
Any suggestions?

**Comment by mikesealey** [comment]
Hey @gaborbatori I've had a look through the list of issues and we seem to already have a report of this (or similar) happening here https://github.com/Budibase/budibase/issues/13827

I'll add this discussion to that issue, please feel free to add any additional information in a comment on that issue (app version, cloud vs self-host, steps to replicate would be useful).

**Reply by gaborbatori** [reply]
Hi, thanks!
It doesn't needed to have a formula column, mine is plain varchar.
App is cloud-based, latest version (2.32.6)

To replicate, You just need an external db (mine is MySQL), have a table, import it to budibase, set one column as display column. In design you set up a dataprovider, a dynamic filter and the table itself. any change on the columns in the table setup throws my display col to the latest. If i directly use the db table, it is perfect, but the dynamic filter will not have any effect. If i set the datasource of the table back to the data provider, filter works, but my display col goes behind...

I attached a screenshot of the whole setup.
Components hierarchy, the table itself, and the setup showing my display col and the data source

<img width="1790" alt="image" src="https://github.com/user-attachments/assets/7b761448-f7f8-4078-bfb5-f855c89cb531">

Thanks!
